src/vue/pages/sale-details/SaleCampaign.vue

```diff
@@ -0,0 +1,324 @@
+<template>
```

Comment:
don't be scared of making lots of small components instead of big ones. Here you have at least 2 widgets that should be split on 3 component:

description-viewer
logo-viewer
state-viewer/widget

Reply:
Splitted `SaleCampaignViewer` component into `SaleLogoViewer`, `SaleDescriptionViewer`, and `SaleStateWidget`components